### PR TITLE
MM-47861 Set paniced jobs to "failed"

### PR DESCRIPTION
**Pull request by noxer**
#### Summary
Jobs `panic`ing during execution didn't update the job status and were stuck "in_progress" forever. This PR makes sure the panic is logged and the job status is updated before re-panicing. The panic is maintained because the rest of the server is unprepared to handle jobs failing like that. A clean restart is preferred.

#### Ticket Link
[MM-47861](https://mattermost.atlassian.net/browse/MM-47861)

#### Release Note
```release-note
Crashing jobs now set the job status to "failed"
```

### CODE REVIEW

File: model/job_test.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package model
```
Comment (agnivade) [inline model/job_test.go:4]: What are these for? Seems unrelated?

**Review by agnivade — APPROVED** [review]
Looks great!

File: model/job_test.go (hunk 1 above)
Comment (noxer) [reply model/job_test.go:4]: I can put them into a separate PR. I originally put the `HandleJobPanic` function in `model/job.go` and noticed that there were no tests for the file. So when writing the test I also tested the rest of the methods as well.

Comment (agnivade) [reply model/job_test.go]: Yeah let's keep this PR scoped to only this task :)

**Comment by noxer** [comment]
Sorry, wrong PR

**Review by isacikgoz — APPROVED** [review]
LGTM, nice one!

**Comment by noxer** [comment]
E2E Test looks good.
